.github/workflows/ci-test.yml

```diff
@@ -132,6 +166,313 @@ jobs:
           exit $success
         env:
           DOCKER_CONTENT_TRUST_REPOSITORY_PASSPHRASE: ${{ secrets.DOCKERHUB_KEY_PASSWORD }}
+      - name: push to novi ecr
+        id: push-to-novi-ecr
+        run: |
+          #push to novi ecr with standard names
+          BRANCH="$CHANGES_TARGET_BRANCH"
+          GIT_REV=$(git rev-parse --short=8 HEAD)
+          echo "::set-output name=head-tag::land_$GIT_REV";
+          aws ecr get-login-password --region ${{ secrets.AWS_REGION }} | \
+          docker login --username AWS --password-stdin "${AWS_ECR_ACCOUNT_URL}"
+          docker/docker_republish.sh -t pre_${BRANCH}_${GIT_REV} -o land_${GIT_REV} -r ${AWS_ECR_ACCOUNT_URL} -d
+
+  need-extra-images:
+    runs-on: ubuntu-latest-xl
+    needs: prepare
+    if: ${{ github.event_name == 'push' && needs.prepare.outputs.build-images == 'true' && needs.prepare.outputs.need-extra-images == 'true' }}
+    outputs:
+      prev-tag: ${{ steps.changes.outputs.build-extra-images }}
+    steps:
+      - name: Get PR base ref
+        id: pr-base-ref
+        uses: actions/github-script@v3
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          result-encoding: string
+          script: |
```

Comment:
if all this is doing is getting the base ref, the changes action already calculates that for both auto/canary prs.   It's needed to figure out the list of changed files.